pkg/registry/certificates/certificates/strategy.go

```diff
@@ -50,6 +51,23 @@ func (csrStrategy) NamespaceScoped() bool {
 	return false
 }
 
+// GetResetFields returns the set of fields that get reset by the strategy
+// and should not be modified by the user.
+func (csrStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
+	fields := map[fieldpath.APIVersion]*fieldpath.Set{
```

Comment:
Certificates seem to be a special case as they are basically immutable after creation.
I added both status and spec to the resetFields as update will reset them.